lib/src/js/compiler.dart

```diff
@@ -0,0 +1,90 @@
+import 'dart:js_util';
+
+import 'package:node_interop/js.dart';
+
+import 'compile.dart';
+import 'compile_options.dart';
+import 'reflection.dart';
+import 'utils.dart';
+
+class Compiler {
+  bool _disposed = false;
+
+  void throwIfDisposed() {
+    if (_disposed) {
+      jsThrow(JsError('Compiler has already been disposed.'));
+    }
+  }
+}
+
+class AsyncCompiler extends Compiler {
+  final Set<Promise> _compilations = {};
```

Comment:
Use a [`FutureGroup`](https://pub.dev/documentation/async/latest/async/FutureGroup-class.html) for this.

Reply:
You can avoid this behavior by wrapping the future: Chain the original Future with a `.catch` that returns the thrown exception, add this wrapped Future to FutureGroup, and returns the original Future to the user. It's a bit ugly but should work as needed.

Reply:
Thanks, I added a `catchError` in bafe182f948ae956b52804835769915f36e163ec. Since we are only using this to determine that all Futures have completed and are returning the compilation results as they come, we ldon't need to worry about the return value.